pkg/network/ebpf/c/tracer.c

```diff
@@ -212,14 +224,64 @@ int kprobe__tcp_close(struct pt_regs *ctx) {
     if (is_protocol_classification_supported()) {
         bpf_map_update_with_telemetry(tcp_close_args, &pid_tgid, &t, BPF_ANY);
     }
+
+    return 0;
+}
+
+SEC("kprobe/tcp_done")
+int kprobe__tcp_done(struct pt_regs *ctx) {
+    struct sock *sk;
+    conn_tuple_t t = {};
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    sk = (struct sock *)PT_REGS_PARM1(ctx);
+    __u64 *failed_conn_pid = NULL;
+    int err = 0;
+
+    bpf_probe_read_kernel_with_telemetry(&err, sizeof(err), (&sk->sk_err));
+    if (err == 0 || !tcp_failed_connections_enabled()) {
```

Comment:
Any need to read the error if failed connections are not enabled?

Reply:
updated